CMakeLists.txt

```diff
@@ -163,35 +159,74 @@ set(GENERAL_PROJECT_SOURCES
     ${PROJECT_SOURCE_DIR}/3rdParty/qxt/qxtglobal.cpp
     ${PROJECT_SOURCE_DIR}/3rdParty/qxt/qxtglobalshortcut.cpp
     # qrc files
-    ${PROJECT_SOURCE_DIR}/3rdParty/qmarkdowntextedit/media.qrc
-    ${PROJECT_SOURCE_DIR}/3rdParty/QSimpleUpdater/etc/resources/qsimpleupdater.qrc
     ${PROJECT_SOURCE_DIR}/src/fonts.qrc
     ${PROJECT_SOURCE_DIR}/src/images.qrc
     ${PROJECT_SOURCE_DIR}/src/styles.qrc)
 
-file(
-  GLOB_RECURSE
-  HEADERS
-  ${PROJECT_SOURCE_DIR}/3rdParty/qautostart/qautostart.h
-  ${PROJECT_SOURCE_DIR}/3rdParty/qmarkdowntextedit/linenumberarea.h
-  ${PROJECT_SOURCE_DIR}/3rdParty/qmarkdowntextedit/markdownhighlighter.h
-  ${PROJECT_SOURCE_DIR}/3rdParty/qmarkdowntextedit/qmarkdowntextedit.h
-  ${PROJECT_SOURCE_DIR}/3rdParty/qmarkdowntextedit/qownlanguagedata.h
-  ${PROJECT_SOURCE_DIR}/3rdParty/qmarkdowntextedit/qplaintexteditsearchwidget.h
-  ${PROJECT_SOURCE_DIR}/3rdParty/QSimpleUpdater/include/*.h
-  ${PROJECT_SOURCE_DIR}/3rdParty/QSimpleUpdater/src/*.h
-  ${PROJECT_SOURCE_DIR}/3rdParty/qxt/*.h
-  ${PROJECT_SOURCE_DIR}/src/*.h)
+set(HEADERS
+    ${PROJECT_SOURCE_DIR}/src/aboutwindow.h
+    ${PROJECT_SOURCE_DIR}/src/allnotebuttontreedelegateeditor.h
+    ${PROJECT_SOURCE_DIR}/src/customapplicationstyle.h
+    ${PROJECT_SOURCE_DIR}/src/customDocument.h
+    ${PROJECT_SOURCE_DIR}/src/customMarkdownHighlighter.h
+    ${PROJECT_SOURCE_DIR}/src/dbmanager.h
+    ${PROJECT_SOURCE_DIR}/src/defaultnotefolderdelegateeditor.h
+    ${PROJECT_SOURCE_DIR}/src/editorsettingsbutton.h
+    ${PROJECT_SOURCE_DIR}/src/elidedlabel.h
+    ${PROJECT_SOURCE_DIR}/src/foldertreedelegateeditor.h
+    ${PROJECT_SOURCE_DIR}/src/labeledittype.h
+    ${PROJECT_SOURCE_DIR}/src/listviewlogic.h
+    ${PROJECT_SOURCE_DIR}/src/mainwindow.h
+    ${PROJECT_SOURCE_DIR}/src/nodedata.h
+    ${PROJECT_SOURCE_DIR}/src/nodepath.h
+    ${PROJECT_SOURCE_DIR}/src/nodetreedelegate.h
+    ${PROJECT_SOURCE_DIR}/src/nodetreemodel.h
+    ${PROJECT_SOURCE_DIR}/src/nodetreeview.h
+    ${PROJECT_SOURCE_DIR}/src/nodetreeview_p.h
+    ${PROJECT_SOURCE_DIR}/src/noteeditorlogic.h
+    ${PROJECT_SOURCE_DIR}/src/notelistdelegateeditor.h
+    ${PROJECT_SOURCE_DIR}/src/notelistdelegate.h
+    ${PROJECT_SOURCE_DIR}/src/notelistmodel.h
+    ${PROJECT_SOURCE_DIR}/src/notelistview.h
+    ${PROJECT_SOURCE_DIR}/src/notelistview_p.h
+    ${PROJECT_SOURCE_DIR}/src/pushbuttontype.h
+    ${PROJECT_SOURCE_DIR}/src/singleinstance.h
+    ${PROJECT_SOURCE_DIR}/src/splitterstyle.h
+    ${PROJECT_SOURCE_DIR}/src/styleeditorwindow.h
+    ${PROJECT_SOURCE_DIR}/src/tagdata.h
+    ${PROJECT_SOURCE_DIR}/src/taglistdelegate.h
+    ${PROJECT_SOURCE_DIR}/src/taglistmodel.h
+    ${PROJECT_SOURCE_DIR}/src/taglistview.h
+    ${PROJECT_SOURCE_DIR}/src/tagpool.h
+    ${PROJECT_SOURCE_DIR}/src/tagtreedelegateeditor.h
+    ${PROJECT_SOURCE_DIR}/src/trashbuttondelegateeditor.h
+    ${PROJECT_SOURCE_DIR}/src/treeviewlogic.h
+    ${PROJECT_SOURCE_DIR}/src/updaterwindow.h
```

Comment:
While you're at it, maybe it would make sense to merge these headers into `GENERAL_PROJECT_SOURCES`?

Reply:
Let's just move all sources into `$GENERAL_PROJECT_SOURCES` (including third-parties), and get rid of `$HEADERS` then.